overlord/snapshotstate/backend/backend.go

```diff
@@ -476,6 +455,52 @@ func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer,
 	return nil
 }
 
+// pathsForSnapshot returns a list of absolute paths under 'snapDir' that should
+// be included in the snapshot (based on what directories exist).
+func pathsForSnapshot(snapDir string, snapshot *client.Snapshot) ([]string, error) {
+	dirExists := func(path string) (bool, error) {
+		exists, isDir, err := osutil.DirExists(path)
+		if err != nil {
+			return false, err
+		}
+
+		if exists && isDir {
+			return true, nil
+		}
+
+		if !exists {
+			logger.Debugf("Not saving %q in snapshot #%d of %q as it is does not exist.", path, snapshot.SetID, snapshot.Snap)
+		} else if !isDir {
+			logger.Noticef("Not saving %q in snapshot #%d of %q as it is not a directory.", path, snapshot.SetID, snapshot.Snap)
+		}
+
+		return false, nil
+	}
+
+	if exists, err := dirExists(snapDir); err != nil {
+		return nil, err
+	} else if !exists {
+		return nil, nil
+	}
```

Comment:
I would remove all of this -- in any case the caller has to handle the case where the list of returned paths is empty (in the  rare case that `snapDir` exist but the dirs for `common` and the revision are both missing).

Reply:
I agree. I considered removing it but then left it so that the refactored code was functionally equivalent to the previous code. AFAICT it's unnecessary so I'll remove it